h2o-r/h2o-package/R/word2vec.R

```diff
@@ -1,108 +1,53 @@
   if (missing(initLearningRate) || !is.numeric(initLearningRate)) stop("`initLearningRate` must be numeric")
   if (missing(epochs) || !is.numeric(epochs)) stop("`epochs` must be numeric")
-  if (!is.H2OFrame(trainingH2OFrame)) invisible(nrow(trainingH2OFrame))  # try to force the eval of the frame
-  if (!is.H2OFrame(trainingH2OFrame)) stop("Could not evaluate `trainingH2OFrame` as an H2OFrame object")
 
-  params <- list(training_frame = h2o.getId(trainingH2OFrame),
-                 wordModel = wordModel,
-                 normModel = normModel,
+  params <- list(training_frame = trainingH2OFrame,
+                 wordModel = "SkipGram",
+                 normModel = "HSM",
                  minWordFreq = minWordFreq,
-                 negSampleCnt = negExCnt,
                  vecSize = vecSize,
                  windowSize = windowSize,
                  sentSampleRate = sentSampleRate,
                  initLearningRate = initLearningRate,
                  epochs = epochs)
 
-  res <- .h2o.__remoteSend(.h2o.__W2V, .params = params)
-  .h2o.__waitOnJob(res$job)
-  dest_key <- .h2o.__remoteSend(paste0(.h2o.__JOBS, "/", res$job))$jobs[[1L]]$dest$name
-  w2vmodel <- h2o.getModel(dest_key)
-  new("H2OW2V", h2o = trainingH2OFrame@conn, key = dest_key, train.data=trainingH2OFrame)
+  .h2o.modelJob("word2vec", params, h2oRestApiVersion = 3)
 }
 
 ##
-## Find Synonyms Using an H2OW2V object
+## Find synonyms using a word2vec model.
 ##
-##  @param word2vec: An H2OW2V model.
-##  @param target: A single word, or a vector of words.
+##  @param word2vec: A word2vec model.
+##  @param target: Single word to find synonyms for.
 ##  @param count: The top `count` synonyms will be returned.
 ##
-h2o.synonym<-
-function(word2vec, target, count) {
-  if (!is(word2vec, "H2OW2V")) stop("`word2vec` must be an H2O word2vec object. See h2o.word2vec")
-  if (missing(target)) stop("`target` must be specified")
-  if (!is.character(target)) stop("`target` must be character")
-  if (missing(count)) stop("`count` must be specified")
-  if (!is.numeric(count)) stop("`count` must be numeric")
-
-  params <- list(key = h2o.getId(word2vec), target=target, cnt=count)
-  if (length(target) == 1L) {
-    res <- .h2o.__remoteSend(word2vec@conn, .h2o.__SYNONYMS, .params = params)
-    fr <- data.frame(synonyms = res$synonyms, cosine.similarity = res$cos_sim)
-    fr[with(fr, order(-cosine.similarity)),]
-  } else {
-    .NotYetImplemented()
-#    vecs <- lapply(target, h2o.transform, word2vec)
-#    vec <- colSums(as.data.frame(vecs))
-#    params$vec <- vec
-#    res <- .h2o.__remoteSend(data@conn, .h2o.__SYNONYMS, params)
-#    return(h2o.getFrame(res$key))
-  }
-}
+h2o.findSynonyms <- function(word2vec, target, count = 20) {
+    if (missing(target)) stop("`target` must be specified")
+    if (!is.character(target)) stop("`target` must be character")
+    if (missing(count)) stop("`count` must be specified")
```

Comment:
this check make default value for `count` redundant as it won't be used, see minimal example:
```r
f = function(x=100) { if(missing(x)) stop("x must be provided"); x }
f()
#Error in f() : x must be provided
```
If this is exported function then some extra check for `length` of `target` and `count` could be added, so it stop if user will provide vector and not scalar value.
Also if exported it is worth to confirm that `word2vec` is object of expected class.